buffer/buffer.go

```diff
@@ -578,16 +634,56 @@ func (b *Buffer) getOffsetArgument(call goja.FunctionCall, argIndex int, bb []by
 		// optional arg that defaults to zero
 		offset = 0
 	} else {
-		panic(errors.NewTypeError(b.r, errors.ErrCodeInvalidArgType, "The \"offset\" argument must be of type number."))
+		panic(b.newArgumentNotNumberTypeError("offset"))
 	}
 
 	if offset < 0 || offset+numBytes > int64(len(bb)) {
-		panic(errors.NewError(b.r, nil, errors.ErrCodedOutOfRange, "The value of \"offset\" %d is out of range", offset))
+		panic(b.newArgumentOutOfRangeError("offset", offset))
 	}
 
 	return offset
 }
 
+func (b *Buffer) getVariableLengthReadArguments(call goja.FunctionCall, bb []byte) (int64, int64) {
+	offset := b.getRequiredIntegerArgument(call, "offset", 0)
+	byteLength := b.getRequiredIntegerArgument(call, "byteLength", 1)
+
+	if byteLength < 1 || byteLength > 6 {
+		panic(b.newArgumentOutOfRangeError("byteLength", byteLength))
+	}
+	if offset < 0 || offset+byteLength > int64(len(bb)) {
+		panic(b.newArgumentOutOfRangeError("offset", offset))
+	}
+
+	return offset, byteLength
+}
+
+func signExtend(value int64, numBytes int64) int64 {
+	// we don't have to turn this to a uint64 first because numBytes < 8 so
+	// the sign bit will never pushed out of the int64 range
+	return (value << (64 - 8*numBytes)) >> (64 - 8*numBytes)
+}
+
+func (b *Buffer) getRequiredIntegerArgument(call goja.FunctionCall, name string, argIndex int) int64 {
+	arg := call.Argument(argIndex)
+	if isNumber(arg) {
+		return arg.ToInteger()
+	}
+	if goja.IsUndefined(arg) {
+		panic(errors.NewTypeError(b.r, errors.ErrCodeInvalidArgType, "The \"%s\" argument is required.", name))
+	}
+
+	panic(b.newArgumentNotNumberTypeError(name))
+}
+
+func (b *Buffer) newArgumentNotNumberTypeError(name string) *goja.Object {
+	return errors.NewTypeError(b.r, errors.ErrCodeInvalidArgType, "The \"%s\" argument must be of type number.", name)
+}
+
+func (b *Buffer) newArgumentOutOfRangeError(name string, v int64) *goja.Object {
+	return errors.NewError(b.r, nil, errors.ErrCodedOutOfRange, "The value of \"%s\" %d is out of range", name, v)
```

Comment:
This should have `RangeError` as a prototype. You can either add `errors.NewRangeError` or just fetch the prototype here and pass it to `NewError`.